Quantum-Control-Applications/Superconducting/Single Fixed Transmon/readout_duration_opt.py

```diff
@@ -0,0 +1,221 @@
+"""
+optimal_weights.py: Optimal weights for the readout pulse
+"""
+from qm.qua import *
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from configuration import *
+import matplotlib.pyplot as plt
+import numpy as np
+from qm import SimulationConfig
+from qualang_tools.loops import from_array
+
+###########
+# Helpers #
+###########
+
+
+def divide_array_in_half(arr):
+    split_index = len(arr) // 2
+    arr1 = arr[:split_index]
+    arr2 = arr[split_index:]
+    return arr1, arr2
+
+
+def create_linear_array_of_same_length(arr):
+    length = len(arr)
+    return [i * 4 for i in range(length)]
+
+
+def create_complex_array(arr1, arr2):
+    return arr1 + 1j * arr2
+
+
+def plot_complex_array(arr):
+    plt.figure()
+    plt.plot(arr.real)
+    plt.plot(arr.imag)
+    plt.show()
+
+
+def subtract_complex_arrays(arr1, arr2):
+    return arr1 - arr2
```

Comment:
This is probably not needed to simplify the program

Reply:
yes